format/ComputeIR.fbs

```diff
@@ -0,0 +1,267 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+include "Schema.fbs";
+
+namespace org.apache.arrow.flatbuf.computeir;
+
+/// Avoid use of org.apache.arrow.Buffer because it requires a
+/// sidecar block of bytes.
+table InlineBuffer {
+  // ulong is used to guarantee alignment and padding of `bytes` so that flatbuffers
+  // and other alignment sensitive blobs can be stored here
+  bytes: [ulong] (required);
```

Comment:
this has implications for languages that aren't C++ (I don't think the ergonmics of converting long[] to byte[] is quite as easy in ava.

Reply:
I think this can be resolved by taking a leaf out of parquet2's book and defining InlineBuffer as a union of vectors of each primitive type, I'll try that.

Reply:
I'm not sure Unions in FB are equivelant to union's in C++ (I don't know that if you select a byte union, you end up sharing the same memory).

Reply:
you might be able to guarantee alignment by having a required long followed by a bytes: [byte] field

Reply:
actually, I think creating a struct { int padding, bytes: byte[] } and making that member should be guaranteed to align.

Reply:
I notice the `ForceVectorAlignment` function in C++

https://github.com/google/flatbuffers/blob/master/include/flatbuffers/flatbuffers.h#L1755

We should look at whether

```
table InlineBuffer {
  padding_for_alignment:ulong = 0 (required);
  bytes:[ubyte] (required);
}
```

guarantees `bytes` to be aligned

Reply:
I've tried an approach inspired by `arrow2`'s decision to include the primitive type being stored in buffers (and thus also alignment information) all the way down to `struct Bytes`

https://github.com/jorgecarleitao/arrow2/blob/main/src/buffer/bytes.rs#L39

I *think* this guarantees alignment without requiring padding fields or reinterpret_casts